### PR TITLE
Snippet Options Need to be Merged into Parent App

### PR DESCRIPTION
**Pull request by mikemunsie**
The previous way we had it didn't allow the snippet options to get merged into the parent apps options. It looked like it should have worked, but instead was merging into the config (when we really needed it in the `app.options`).

I noticed after the options are now merged, Snippets don't work for `.md` files. Works great if you have `.hbs` files, but something is going on where it's not capturing the snippet parts after transforming .md -> .hbs. I'd love to add that in this PR, but I'm unable to find a reason why that's not working right now.

EDIT: I found the source of why live-demo / snippets weren't working with`.md` files and have added that to this PR.  I have also learned that `ember-code-snippet` by default does not support .md files and have created a PR on their side for that.

### CODE REVIEW

**Comment by ming-codes** [comment]
This solves #18 right?

**Comment by mikemunsie** [comment]
@ming-codes Yep! It should take care of it.  I got the inspiration from a few addons where they used the `include` function to modify app options. See: https://github.com/ember-cli/ember-cli-htmlbars-inline-precompile/blob/master/index.js#L50

File: index.js [hunk 1]
```diff
@@ -64,10 +66,8 @@ module.exports = {
     ].filter(Boolean));
   },
 
-  treeForPublic() {
-    let parentAddon = this.parent.findAddonByName(this.parent.name());
-    if (!parentAddon) { return; }
-
+  treeForPublic() { 
+    let parentAddon = this.parent.treePaths ? this.parent : this;
```
Comment (dfreeman) [inline index.js]: I'm not sure I follow how this is related to the snippets change?

File: index.js [hunk 2]
```diff
@@ -28,17 +28,19 @@ module.exports = {
...
-        end: /{{\/docs-snippet}}/,
-      },
+      }
     };
   },
 
-  included() {
+ included(parentApp) {
     this._super.included.apply(this, arguments);
+    parentApp.options = Object.assign(parentApp.options, {
```
Comment (dfreeman) [inline index.js]: This is going to blow away the parent's own `snippetSearchPaths` and `snippetRegexes` options if they happen to have them set. Can we avoid that?

File: index.js (hunk 1 above)
Comment (mikemunsie) [inline index.js]: You're absolutely right. This is more related to when pulling `ember-cli-addon-docs` into your project, you get a `this.parent.name` is not a function and the project wouldn't run. 

```
this.parent.name is not a function
TypeError: this.parent.name is not a function
    at Class.treeForPublic (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli-addon-docs/index.js:68:63)
    at Class._treeFor (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/models/addon.js:551:33)
    at Class.treeFor (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/models/addon.js:511:21)
    at addons.map.addon (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/models/addon.js:383:34)
    at Array.map (native)
    at Class.eachAddonInvoke (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/models/addon.js:381:24)
    at Class.treeFor (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/models/addon.js:510:22)
    at project.addons.map.addon (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/broccoli/ember-app.js:557:22)
    at Array.map (native)
    at EmberAddon.addonTreesFor (/Users/munsie/Documents/repos/testing-ui/node_modules/ember-cli/lib/broccoli/ember-app.js:555:32)
```

Comment (dfreeman) [inline index.js]: I actually just landed a fix to give a more descriptive error for that case 🙂 
https://github.com/ember-learn/ember-cli-addon-docs/commit/bc187faae8c333ae4dda94eba5058a480863c154

Comment (mikemunsie) [inline index.js]: Ha! Yes, ironically I forgot to move that into my devDeps. Whoops! Reverting the index.js file now. Good catch 💯

**Comment by mikemunsie** [comment]
Alright, so I embarassingly put ember-cli-addon-docs as dependency when it was meant to be a devDependency. I'm gonna close this PR for now, and clean things up a bit. Sorry for the mess! Learned some stuff though 👍

**Pull request closed without merging**